src/components/editors/BoxLinksEditor.tsx

```diff
@@ -0,0 +1,132 @@
+ * limitations under the License.
+ ***************************************************************************** */
+
+import { observer } from "mobx-react-lite";
+import { useCallback, useMemo, useRef, useState } from "react";
+import { createUseStyles } from "react-jss";
+import { useBoxesStore } from "../../hooks/useBoxesStore";
+import { useDictionaryLinksStore } from "../../hooks/useDictionaryLinksStore";
+import useOutsideClickListener from "../../hooks/useOutsideClickListener";
+import { useSelectedDictionaryStore } from "../../hooks/useSelectedDictionaryStore";
+import { DictionaryRelation } from "../../models/Dictionary";
+import Icon from "../Icon";
+import Select from "../util/Select";
+import { useLinksStyles } from "./DictionaryLinksEditor";
+
+interface DictionaryLinkProps {
+	link: DictionaryRelation;
+	deleteLink: () => void;
+}
+
+const useLinkStyle = createUseStyles({
+	link: {
+		cursor: 'pointer',
+		padding: '0 2px',
+		display: 'flex',
+		justifyContent: 'space-between',
+		alignItems: 'center',
+		'&:hover': {
+			backgroundColor: '#e5e5e5'
+		}
+	},
+	title: {
+		display: 'flex',
+		'&>p': {
+			margin: '0 0 0 2px'
+		}
+	},
+	delete: {
+		display: 'inline-flex',
+		padding: '0 2px',
+		verticalAlign: 'middle',
+		backgroundColor: 'transparent',
+		outline: 'none',
+		border: 'none',
+		cursor: 'pointer',
+	}
+})
+
+const Link = ({link, deleteLink}: DictionaryLinkProps) => {
+	const classes = useLinkStyle();
+	return (
+		<div className={classes.link}>
+			<div className={classes.title}>
+				<Icon id='box' stroke='black' />
+				<p>{link.box}</p>
+			</div>
+			<button className={classes.delete} onClick={deleteLink}><Icon id='cross' stroke='black' width={8} height={8}/></button>
+		</div>
+	)
+}
+
+const BoxLinksEditor = () => {
+	const classes = useLinksStyles();
+	const boxesStore = useBoxesStore();
+	const selectedDictionaryStore = useSelectedDictionaryStore();
+	const dictionaryLinksStore = useDictionaryLinksStore();
+
+	const options = useMemo(() => {
+		return boxesStore.boxes
+			.filter(box => !dictionaryLinksStore.linkedBoxes?.some(link => link.box === box.name))
+			.map(box => box.name)
+	}, [boxesStore.boxes, dictionaryLinksStore.linkedBoxes]);
+
+	const [newLinkedBoxName, setNewLinkedBoxName] = useState(options[0]);
+	const [showAddBox, setShowAddBox] = useState(false);
+
+	const ref = useRef<HTMLDivElement>(null);
+	useOutsideClickListener(ref, () => {
+		setShowAddBox(false);
+	})
+
+	const applyNewLink = useCallback(() => {
+		setShowAddBox(false)
+		if (selectedDictionaryStore.dictionary) {
+			const newLinkDictionary: DictionaryRelation = {
+				name: `${newLinkedBoxName}-dictionary`,
+				box: newLinkedBoxName,
+				dictionary: {
+					name: selectedDictionaryStore.dictionary.name,
+					type: 'MAIN'
+				}
+			}
+			dictionaryLinksStore.addLinkDictionary(newLinkDictionary);
+		}
+	}, [newLinkedBoxName, selectedDictionaryStore.dictionary])
+
+	return dictionaryLinksStore.linkedBoxes 
```

Comment:
dictionaryLinksStore.linkedBoxes is an array, and it always will be truthy, so why do we need this check?